packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -1379,14 +1401,51 @@ export class GarbageCollector implements IGarbageCollector {
                 );
             } else {
                 nodeStateTracker.updateTracking(currentReferenceTimestampMs);
-                if (this.tombstoneMode && nodeStateTracker.state === UnreferencedState.SweepReady) {
-                    const nodeType = this.runtime.getNodeType(nodeId);
-                    if (nodeType === GCNodeType.DataStore || nodeType === GCNodeType.Blob) {
-                        this.tombstones.push(nodeId);
-                    }
+                if (nodeStateTracker.state === UnreferencedState.SweepReady) {
+                    sweepReadyNodes.push(nodeId);
                 }
             }
         }
+
+        return sweepReadyNodes;
+    }
+
+    private validateSweepRoutes(sweepRoutes: string[], allRoutes: string[]) {
+        const deletedRoutes: Set<string> = new Set(sweepRoutes);
+        const notDeletedRoutes = allRoutes.filter((route) => !sweepRoutes.includes(route));
+        // Verify that if the datastore is deleted, all its sub datastore routes are deleted as well
+        for (const route of notDeletedRoutes) {
+            const pathParts = route.split("/");
+            if (pathParts.length >= 2) {
+                assert(!deletedRoutes.has(`/${pathParts[1]}`), "Sub node routes should be deleted when deleting the node!");
+            }
+        }
+    }
+
+    /**
+     * Removes nodes that are swept from GC state and updates the deleted nodes
+     */
+    private updateSweptNodeState(gcNodes: string[]) {
+        for (const nodeId of gcNodes) {
+            const nodeStateTracker = this.unreferencedNodesState.get(nodeId);
+            if (nodeStateTracker !== undefined) {
+                // Stop tracking so as to clear out any running timers.
+                nodeStateTracker.stopTracking();
+                // Delete the node as we don't need to track it any more.
+                this.unreferencedNodesState.delete(nodeId);
+            }
+        }
+
+        const sweptNodes = new Set<string>(gcNodes);
+        assert(this.gcDataFromLastRun !== undefined, "GC data should be defined when we are running sweep");
+        const remainingNodes: { [ id: string ]: string[]; } = {};
+        Object.entries(this.gcDataFromLastRun.gcNodes).forEach(([nodeId, routes]) => {
```

Comment:
It's probably simpler to just iterate over `gcNodes` and for each nodeId in it, simply call this.gcDataFromLastRun.delete[nodeId[.

Reply:
There's no specific delete function that I know of. The delete that I do know is `delete this.gcDataFromLastRun.gcNodes[key]` which is not recommended

Reply:
yeah, that's what I meant. I didn't know that it was not recommeneded.

Reply:
I'm fine either way, it doesn't matter to me

Reply:
`@typescript-eslint/no-dynamic-delete` is the problem

Reply:
I see. I didn't know about that. It doesn't matter to me either :-). I was just thinking from the point of view of simplicity but if its not recommended, we could use the other way.

Reply:
Also, if you like the suggestion in this comment https://github.com/microsoft/FluidFramework/pull/13586#discussion_r1089277705, you simply skip adding deleted nodes when update this.gcDataFromLastRun in updateCurrentState.

Reply:
Yeah, I'll do that, I like that